docs/fastdds/transport/datasharing.rst

```diff
@@ -0,0 +1,213 @@
+.. include:: ../../03-exports/aliases.include
+.. include:: ../../03-exports/aliases-api.include
+
+.. _datasharing-delivery:
+
+Data-sharing delivery
+=====================
+
+*Fast DDS* allows to speed up communications between entities within the same machine
+by sharing the history of the |DataWriter| with the |DataReader| through shared memory,
+thus avoiding any of the overhead involved in the transport layer,
+and effectively avoiding any data copy between DataWriter and DataReader.
+
+.. note::
+    Although Data-sharing delivery uses shared memory,
+    it differs from :ref:`transport_sharedMemory_sharedMemory`
+    in that :ref:`transport_sharedMemory_sharedMemory` is a full-compliant transport.
+    That means that with :ref:`transport_sharedMemory_sharedMemory`
+    the data being transmitted must be copied from the DataWriter history to the transport
+    and from the transport to the DataReader.
+    With Data-sharing these copies can be avoided
+
+The figure below shows a comparison between the different transports available in *Fast DDS*.
+
+.. figure:: /01-figures/fast_dds/transport/transport_comparison.png
+    :align: center
+
+.. contents::
```

Comment:
I will move this index above the figure. Otherwise is hidden and in a weird position.